### PR TITLE
hw07 done

### PR DESCRIPTION
**Pull request by emineyvazov222**
there may be mistakes, but I tried to do what I know
@sabuhihuseynov

### CODE REVIEW

File: src/main/java/org/spring/FullVersion/Human.java [hunk 1]
```diff
@@ -82,12 +97,12 @@ public void setSurname(String surname) {
         this.surname = surname;
     }
 
-    public int getYear() {
-        return year;
+    public long getYear() {
+        return birthDate;
     }
 
     public void setYear(int year) {
```
Comment (sabuhihuseynov) [inline src/main/java/org/spring/FullVersion/Human.java]: also change method names pls

File: src/main/java/org/spring/FullVersion/Human.java [hunk 2]
```diff
@@ -82,12 +97,12 @@ public void setSurname(String surname) {
         this.surname = surname;
     }
 
-    public int getYear() {
-        return year;
+    public long getYear() {
```
Comment (sabuhihuseynov) [inline src/main/java/org/spring/FullVersion/Human.java]: also change method names pls

File: src/main/java/org/spring/FullVersion/Human.java [hunk 3]
```diff
@@ -1,44 +1,59 @@
...
+        this.surname = surname;
+        this.iq = iq;
+        this.birthDate = convertToMillis(birthDateString);
+    }
+
+    private long convertToMillis(String birthDateString) throws ParseException {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
+        Date date = dateFormat.parse(birthDateString);
+        return date.getTime();
+
```
Comment (sabuhihuseynov) [inline src/main/java/org/spring/FullVersion/Human.java:60]: pls look LocalDate, Period classes for converting and describing age

File: src/main/java/org/spring/FullVersion/Human.java [hunk 4]
```diff
@@ -100,19 +115,30 @@ public boolean equals(Object o) {
...
     public int hashCode() {
-        return Objects.hash(name, surname, year, iq, family, schedule);
+        return Objects.hash(name, surname, birthDate, iq, family, schedule);
+    }
+
+    public String describeAge() {
+        long currentMillis = System.currentTimeMillis();
+        long ageInMillis = currentMillis - birthDate;
+
+        long years = TimeUnit.MILLISECONDS.toDays(ageInMillis) / 365;
```
Comment (sabuhihuseynov) [inline src/main/java/org/spring/FullVersion/Human.java]: pls investigate Period class

File: src/main/java/org/spring/FullVersion/FamilyService.java [hunk 5]
```diff
@@ -79,7 +79,7 @@ public Family adoptChild(Family family, Human child) {
 
     public void deleteAllChildrenOlderThen(int age) {
         for (Family family : familyDao.getAllFamilies()) {
-            family.getChildren().removeIf(child -> child.getAge() > age);
+            family.getChildren().removeIf(child -> Integer.parseInt(child.describeAge()) > age);
```
Comment (sabuhihuseynov) [inline src/main/java/org/spring/FullVersion/FamilyService.java]: how it can parse based on describeAge method response?

**Review by sabuhihuseynov — CHANGES_REQUESTED** [review]
pls look

File: src/main/java/org/spring/FullVersion/FamilyService.java [hunk 6]
```diff
@@ -78,10 +81,20 @@ public Family adoptChild(Family family, Human child) {
...
     public void deleteAllChildrenOlderThen(int age) {
-        for (Family family : familyDao.getAllFamilies()) {
-            family.getChildren().removeIf(child -> Integer.parseInt(child.describeAge()) > age);
-            familyDao.saveFamily(family);
-        }
+        LocalDate currentDate = LocalDate.now();
+
+        familyDao.getAllFamilies().stream()
+                .peek(family -> family.getChildren().removeIf(child -> {
+                    LocalDate birthDate = convertToLocalDate(child.describeAge());
```
Comment (sabuhihuseynov) [inline src/main/java/org/spring/FullVersion/FamilyService.java]: instead of writing this code here you can implement getAge method on Human class

**Review by sabuhihuseynov — CHANGES_REQUESTED** [review]
pls look

File: src/main/java/org/spring/FullVersion/FamilyService.java (hunk 6 above)
Comment (emineyvazov222) [reply src/main/java/org/spring/FullVersion/FamilyService.java]: like child.getAge? @sabuhihuseynov

File: src/main/java/org/spring/FullVersion/FamilyService.java [hunk 7]
```diff
@@ -85,7 +85,7 @@ public void deleteAllChildrenOlderThen(int age) {
 
         familyDao.getAllFamilies().stream()
                 .peek(family -> family.getChildren().removeIf(child -> {
-                    LocalDate birthDate = convertToLocalDate(child.describeAge());
+                    LocalDate birthDate = convertToLocalDate(Integer.toString(child.getAge()));
```
Comment (sabuhihuseynov) [inline src/main/java/org/spring/FullVersion/FamilyService.java]: you already get age from child.getAge() why you parse it to string and send convertToLocalDate?

**Review by sabuhihuseynov — CHANGES_REQUESTED** [review]
pls look

File: src/main/java/org/spring/FullVersion/FamilyService.java (hunk 7 above)
Comment (emineyvazov222) [reply src/main/java/org/spring/FullVersion/FamilyService.java]: LocalDate birthDate = LocalDate.ofEpochDay(child.getAge()); 
Do you want it in this form? @sabuhihuseynov

**Review by sabuhihuseynov — APPROVED** [review]

**Pull request merged**